### PR TITLE
moved hash.rs to bigint library

### PR DESCRIPTION
**Pull request by debris**
in next pr, I will make `ethkey`/`ethstore` use bigint library

### CODE REVIEW

File: ethcore/src/client/client.rs [hunk 1]
```diff
@@ -648,7 +648,7 @@ impl BlockChainClient for Client {
 			timestamp: view.timestamp(),
 			difficulty: view.difficulty(),
 			last_hashes: last_hashes,
-			gas_used: U256::zero(),
+			gas_used: Default::default(),
```
Comment (tomusdrw) [inline ethcore/src/client/client.rs]: TBH I prefer more explicit approach in such cases (`U256::default(), U256::from(0)`), for me it just means less cognitive load.

Look at `transaction_hash` below especialy, It's quite difficult to understand what type is returned (it gets even more tricky when there is some more `.map`ing on options/results/vectors done).

But it's just an opinion :) I'm good with the changes :+1:

Comment (NikVolf) [inline ethcore/src/client/client.rs]: yes i think U256::zero() is much more preferred even
we deal with numbers here, not with some objects/classes

File: util/bigint/src/hash.rs [hunk 2]
```diff
@@ -605,25 +610,26 @@ mod tests {
 	}
 
 	#[test]
+	#[ignore]
 	fn shift_bloomed() {
-		use sha3::Hashable;
+		//use sha3::Hashable;
```
Comment (tomusdrw) [inline util/bigint/src/hash.rs]: What happened here? Ignored and commented out?

Comment (debris) [inline util/bigint/src/hash.rs]: I should have mentioned that in PR description. It's commented out and will be moved out of `hash.rs`. These tests use keccak function and I didn't want to bring sha3 as a dependency here.

**Comment by debris** [comment]
blooms methods are moved to separate trait - `Bloomable`

**Comment by rphmeier** [comment]
needs a merge -- the `signature_to_rsv` stuff is a little less pretty than before. maybe the newtype pattern would help there.

File: util/network/src/host.rs [hunk 3]
```diff
@@ -1150,7 +1150,7 @@ fn key_save_load() {
 #[test]
 fn host_client_url() {
 	let mut config = NetworkConfiguration::new();
-	let key = h256_from_hex("6f7b0d801bc7b5ce7bbd930b84fd0369b3eb25d09be58d64ba811091046f3aa2");
+	let key = "6f7b0d801bc7b5ce7bbd930b84fd0369b3eb25d09be58d64ba811091046f3aa2".into();
```
Comment (gavofyork) [inline util/network/src/host.rs]: why is this a simple `.into()` whereas the address stuff requires a `.parse().unwrap()`?

File: ethcore/src/ethereum/mod.rs [hunk 4]
```diff
@@ -69,12 +69,12 @@ mod tests {
 		let mut db = db_result.take();
 		spec.ensure_db_good(db.as_hashdb_mut()).unwrap();
 		let s = State::from_existing(db, genesis_header.state_root.clone(), engine.account_start_nonce(), Default::default()).unwrap();
-		assert_eq!(s.balance(&address_from_hex("0000000000000000000000000000000000000001")), U256::from(1u64));
-		assert_eq!(s.balance(&address_from_hex("0000000000000000000000000000000000000002")), U256::from(1u64));
-		assert_eq!(s.balance(&address_from_hex("0000000000000000000000000000000000000003")), U256::from(1u64));
-		assert_eq!(s.balance(&address_from_hex("0000000000000000000000000000000000000004")), U256::from(1u64));
-		assert_eq!(s.balance(&address_from_hex("102e61f5d8f9bc71d0ad4a084df4e65e05ce0e1c")), U256::from(1u64) << 200);
-		assert_eq!(s.balance(&address_from_hex("0000000000000000000000000000000000000000")), U256::from(0u64));
+		assert_eq!(s.balance(&"0000000000000000000000000000000000000001".parse().unwrap()), U256::from(1u64));
```
Comment (gavofyork) [inline ethcore/src/ethereum/mod.rs]: `.parse().unwrap()` -> `.into()`?

**Comment by gavofyork** [comment]
looks good (particularly the `.into()` refactoring) except for the errant `.parse().unwrap()`s.

File: ethcore/src/ethereum/mod.rs (hunk 4 above)
Comment (rphmeier) [inline ethcore/src/ethereum/mod.rs]: probably would need to be an `.into::<Address>()` as the `balance` function here would accept anything which is `Deref<Target=Address>`

Comment (debris) [inline ethcore/src/ethereum/mod.rs]: I added `From<&'static str>` implementation for hashes some time ago... I forgot to use it here :)

**Pull request merged**